runtime/queries/column_numeric_histogram.go

```diff
@@ -196,5 +198,177 @@ func (q *ColumnNumericHistogram) Resolve(ctx context.Context, rt *runtime.Runtim
 	}
 
 	q.Result = histogramBins
+
+	return nil
+}
+
+func (q *ColumnNumericHistogram) calculateDiagnosticMethod(ctx context.Context, rt *runtime.Runtime, instanceID string, priority int) error {
+	olap, err := rt.OLAP(ctx, instanceID)
+	if err != nil {
+		return err
+	}
+
+	if olap.Dialect() != drivers.DialectDuckDB {
+		return fmt.Errorf("not available for dialect '%s'", olap.Dialect())
+	}
+
+	sanitizedColumnName := safeName(q.ColumnName)
+	minMaxSQL := fmt.Sprintf(
+		`
+			SELECT
+				min(%[2]s) as min,
+				max(%[2]s) as max,
+				max(%[2]s) - min(%[2]s) as range
+			FROM %[1]s 
+			WHERE %[2]s IS NOT NULL
+		`,
+		safeName(q.TableName),
+		sanitizedColumnName,
+	)
+
+	minMaxRow, err := olap.Execute(ctx, &drivers.Statement{
+		Query:    minMaxSQL,
+		Priority: priority,
+	})
+	if err != nil {
+		return err
+	}
+
+	var min, max, rng float64
+	if minMaxRow.Next() {
+		err = minMaxRow.Scan(&min, &max, &rng)
+		if err != nil {
+			minMaxRow.Close()
+			return err
+		}
+	}
+
+	minMaxRow.Close()
+
+	ticks := 40.0
+	if rng < ticks {
+		ticks = rng
+	}
+	niceResult := NiceAndStep(min, max, ticks)
+	startTick := niceResult[0]
+	endTick := niceResult[1]
+	gap := niceResult[2]
```

Comment:
Refactor func to return three args: `startTick, endTick, gap := NiceAndStep(min, max, ticks)`